SteakBot.Core/Services/QuotingService.cs

```diff
@@ -0,0 +1,255 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Discord;
+using Discord.WebSocket;
+
+namespace SteakBot.Core.Services
+{
+    public class QuotingService
+    {
+        public bool TryGetGuild(IDiscordClient discordClient, ulong guildId, out SocketGuild guild)
+        {
+            guild = discordClient.GetGuildAsync(guildId).Result as SocketGuild;
+            return guild != null;
+        }
+
+        public bool TryGetChannel(SocketGuild guild, ulong channelId, out SocketTextChannel channel)
+        {
+            foreach (var socketGuildChannel in guild.Channels)
+            {
+                if (socketGuildChannel is SocketTextChannel tmpChannel && tmpChannel.Id == channelId)
+                {
+                    channel = tmpChannel;
+                    return true;
+                }
+            }
+
+            channel = null;
+            return false;
+        }
+
+        public bool TryGetChannel(SocketGuild guild, string channelMention, out SocketTextChannel channel)
+        {
+            foreach (var socketGuildChannel in guild.Channels)
+            {
+                if (socketGuildChannel is SocketTextChannel tmpChannel && tmpChannel.Mention == channelMention)
+                {
+                    channel = tmpChannel;
+                    return true;
+                }
+            }
+
+            channel = null;
+            return false;
+        }
+
+        public bool TryGetMessage(SocketGuild guild, IUser currentUser, ulong messageId, out IMessage message, SocketTextChannel preferredChannel = null)
+        {
+            var channels = new List<SocketTextChannel>(guild.TextChannels);
+            if (preferredChannel != null)
+            {
+                channels.Insert(0, preferredChannel);
+            }
+
+            foreach (var channel in channels)
+            {
+                if (!channel.Users.Any(x => x.Username == currentUser.Username && x.DiscriminatorValue == currentUser.DiscriminatorValue))
+                {
+                    continue;
+                }
+
+                message = channel.GetMessageAsync(messageId).Result;
+                if (message != null)
+                {
+                    return true;
+                }
+            }
+
+            message = null;
+            return false;
+        }
+
+        public async Task<List<IMessage>> GetMessageList(SocketTextChannel channel, ulong firstMessageId, ulong lastMessageId)
+        {
+            var firstMessage = await channel.GetMessageAsync(firstMessageId);
+            if (firstMessage == null)
+            {
+                return new List<IMessage>();
+            }
+
+            var downloadedMessages = channel.GetMessagesAsync(firstMessageId, Direction.After, 25);
+            var firstPage = (await downloadedMessages.First()).ToArray();
+
+            var messages = new List<IMessage>
```

Comment:
You can specify the size of the List collection `firstPage.Length  + 1` (+1 is for the `firstMessage`) that is added in the initializer if its not included in the `firstPage.Length`.

Reply:
Done.